library/src/main/java/com/novoda/downloadmanager/Optional.java

```diff
@@ -0,0 +1,62 @@
+package com.novoda.downloadmanager;
+
+public final class Optional<T> {
```

Comment:
Had to introduce an `Optional` for some of the recommended changes. Now we can remove `@Nullable` from the codebase.